x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/phases/frozen_phase.tsx

```diff
@@ -0,0 +1,210 @@
+
+import {
+  LearnMoreLink,
+  ActiveBadge,
+  PhaseErrorMessage,
+  OptionalLabel,
+  ErrableFormRow,
+  MinAgeInput,
+  NodeAllocation,
+  SetPriorityInput,
+} from '../components';
+
+const freezeLabel = i18n.translate('xpack.indexLifecycleMgmt.frozenPhase.freezeIndexLabel', {
+  defaultMessage: 'Freeze index',
+});
+
+const frozenProperty: keyof Phases = 'frozen';
+const phaseProperty = (propertyName: keyof FrozenPhaseInterface) => propertyName;
+
+interface Props {
+  setPhaseData: (key: keyof FrozenPhaseInterface & string, value: string | boolean) => void;
+  phaseData: FrozenPhaseInterface;
+  isShowingErrors: boolean;
+  errors?: PhaseValidationErrors<FrozenPhaseInterface>;
+  hotPhaseRolloverEnabled: boolean;
+}
+export class FrozenPhase extends PureComponent<Props> {
+  render() {
+    const {
+      setPhaseData,
+      phaseData,
+      errors,
+      isShowingErrors,
+      hotPhaseRolloverEnabled,
+    } = this.props;
+
+    return (
+      <div id="frozenPhaseContent" aria-live="polite" role="region">
+        <EuiDescribedFormGroup
+          title={
+            <div>
+              <h2 className="eui-displayInlineBlock eui-alignMiddle">
+                <FormattedMessage
+                  id="xpack.indexLifecycleMgmt.editPolicy.frozenPhase.frozenPhaseLabel"
+                  defaultMessage="Frozen phase"
+                />
+              </h2>{' '}
+              {phaseData.phaseEnabled && !isShowingErrors ? <ActiveBadge /> : null}
+              <PhaseErrorMessage isShowingErrors={isShowingErrors} />
+            </div>
+          }
+          titleSize="s"
+          description={
+            <Fragment>
+              <p>
+                <FormattedMessage
+                  id="xpack.indexLifecycleMgmt.editPolicy.frozenPhase.frozenPhaseDescriptionText"
+                  defaultMessage="You are querying your index very infrequently, so you can allocate shards
+                  on the least performant hardware.
+                  Because your queries are slower, you can reduce the number of replicas."
+                />
+              </p>
+              <EuiSwitch
+                data-test-subj="enablePhaseSwitch-frozen"
+                label={
+                  <FormattedMessage
+                    id="xpack.indexLifecycleMgmt.editPolicy.frozenPhase.activateWarmPhaseSwitchLabel"
+                    defaultMessage="Activate frozen phase"
+                  />
+                }
+                id={`${frozenProperty}-${phaseProperty('phaseEnabled')}`}
+                checked={phaseData.phaseEnabled}
+                onChange={(e) => {
+                  setPhaseData(phaseProperty('phaseEnabled'), e.target.checked);
+                }}
+                aria-controls="frozenPhaseContent"
+              />
+            </Fragment>
+          }
+          fullWidth
+        >
+          <Fragment>
+            {phaseData.phaseEnabled ? (
+              <Fragment>
+                <MinAgeInput<FrozenPhaseInterface>
```

Comment:
`MinAgeInput`'s props are currently typed to accept warm, cold and delete phases. I'm not sure why the type check is not picking it up, but it might be good to add frozen phase to props generics.